pkg/thanos/statefulset.go

```diff
@@ -303,13 +322,39 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 			watchedDirectories = append(watchedDirectories, mountPath)
 		}
 
+		if version.GTE(semver.MustParse("0.21.0")) {
+			var fields monitoringv1.WebConfigFileFields
+			if tr.Spec.Web != nil {
+				fields = tr.Spec.Web.WebConfigFileFields
+			}
+
+			webConfig, err := webconfig.New(webConfigDir, webConfigSecretName(tr.Name), fields)
+			if err != nil {
+				return nil, err
+			}
+
+			confArg, configVol, configMount, err := webConfig.GetMountParameters()
+			if err != nil {
+				return nil, err
+			}
+			containerArgs = append(containerArgs, fmt.Sprintf("--http.config=%s", confArg.Value))
+			trVolumes = append(trVolumes, configVol...)
+			trVolumeMounts = append(trVolumeMounts, configMount...)
+
+			if tr.Spec.Web != nil {
+				configReloaderWebConfigFile = confArg.Value
+				configReloaderVolumeMounts = append(configReloaderVolumeMounts, configMount...)
+			}
```

Comment:
The test exists in Alertmanager/Prometheus to deal with older config reloader versions. Here we can expect that users have already updated to a recent version.
```suggestion
			configReloaderWebConfigFile = confArg.Value
			configReloaderVolumeMounts = append(configReloaderVolumeMounts, configMount...)